### PR TITLE
Add ADO pipeline preview run feature and test 

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

Added a feature in the image builder that allows ADO (Azure DevOps) pipeline to run in preview mode. This allows users to see the final YAML of the pipeline before executing it. The use of this option can be flagged with adoPreviewRun and it has been limited to work only when running in ADO and not locally. Made companion changes to the tests for these features ensuring all new code is covered. This addition was made to aid debug and development efforts by providing more comprehensive information about pipeline executions in ADO.

**Related issue(s)**
Depends on #9461 
See #9237

### CODE REVIEW

**Comment by dekiel** [comment]
Example response from preview run.
`Image build failed with error: build in ADO failed, failed running ADO pipeline, err: /azure-pipelines.yml (Line: 117, Col: 15): Key not found 'CI'exit status 1`
Preview run was able to detect I'm referencing non existing pipeline parameter in a task.

File: pkg/azuredevops/pipelines/pipelines_test.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 	"strings"
 
 	"github.com/kyma-project/test-infra/pkg/azuredevops/pipelines"
-	"github.com/kyma-project/test-infra/pkg/azuredevops/pipelines/mocks"
+	pipelinesmocks "github.com/kyma-project/test-infra/pkg/azuredevops/pipelines/mocks"
```
Comment (akiioto) [inline pkg/azuredevops/pipelines/pipelines_test.go:16]:
```go
pipelinesMocks
```

Comment (dekiel) [reply pkg/azuredevops/pipelines/pipelines_test.go]: Corrected

File: cmd/image-builder/main.go [hunk 2]
```diff
@@ -201,10 +203,12 @@ func prepareADOTemplateParameters(options options) (adopipelines.OCIImageBuilder
 	}
 
 	number, present := os.LookupEnv("PULL_NUMBER")
-	if !present {
+	if jobType == "presubmit" && !present {
```
[the remark this answers is not in the document]
Comment (dekiel) [reply cmd/image-builder/main.go]: Changed names.

**Review by akiioto — APPROVED** [review]

**Comment by dekiel** [comment]
/override pull-build-githubrelease
This prowjobs is not valid for this. pull request.

**Pull request merged**